src/nylas.ts

```diff
@@ -130,38 +132,36 @@ class Nylas {
       throw new Error('exchangeCodeForToken() must be called with a code');
     }
 
-    return new Promise<string>((resolve, reject) => {
-      const options = {
-        method: 'GET',
-        json: true,
-        url: `${this.apiServer}/oauth/token`,
-        qs: {
-          client_id: this.clientId,
-          client_secret: this.clientSecret,
-          grant_type: 'authorization_code',
-          code: code,
-        },
-      };
-
-      return request(options, (error, response, body) => {
-        if (!body || !body['access_token']) {
-          let errorMessage = 'No access token in response';
-          if (body && body.message) errorMessage = body.message;
-          error = error ? error : new Error(errorMessage);
+    const url = new URL(`${this.apiServer}/oauth/token`);
+    url.searchParams.set('client_id', this.clientId);
+    url.searchParams.set('client_secret', this.clientSecret);
+    url.searchParams.set('grant_type', 'authorization_code');
+    url.searchParams.set('code', code);
+    return fetch(url)
+      .then(response => {
+        if (response.ok) {
+          return response.json();
         }
-        if (error) {
-          reject(error);
+      })
+      .then(
+        body => {
+          if (!body || !body['access_token']) {
+            let errorMessage = 'No access token in response';
+            if (body && body.message) errorMessage = body.message;
+            throw new Error(errorMessage);
+          }
           if (callback) {
-            return callback(error);
+            callback(null, body['access_token']);
           }
-        } else {
-          resolve(body['access_token']);
+          return body['access_token'];
+        },
+        error => {
           if (callback) {
-            return callback(null, body['access_token']);
+            callback(error);
           }
+          throw error;
```

Comment:
wouldn't it make sense to recreate an Error so it includes current frame in a stacktrace?

Reply:
Done